api/units.go

```diff
@@ -101,15 +101,33 @@ func (ur *unitsResource) set(rw http.ResponseWriter, req *http.Request, item str
 		return
 	}
 
+	newUnit := false
 	if eu == nil {
 		if len(su.Options) == 0 {
 			err := errors.New("unit does not exist and options field empty")
 			sendError(rw, http.StatusConflict, err)
+			return
 		} else if err := ValidateOptions(su.Options); err != nil {
 			sendError(rw, http.StatusBadRequest, err)
+			return
 		} else {
-			ur.create(rw, su.Name, &su)
+			newUnit = true
 		}
+	} else if eu.Name == su.Name && len(su.Options) > 0 {
+		// There is already a unit with the same name
+		// check the hashes if they do not match then we will
+		// create a new unit with the same name and later
+		// the job will be updated to this new unit.
```

Comment:
> Calm down, I was merely asking for some clarification. The code comment doesn't really make it clear to me how this is supposed to work. I guess that's because I don't know the fleet code and it's etcd interactions by heart -- but then, I think comments should also make sense to people who don't... Those who know it by heart probably don't even need comments.

It's ok, the whole logic was explained in these two patches and the code is commented:
https://github.com/coreos/fleet/pull/1509/commits/0e675d9dca2ebd42d296bde75466e2e49b6daa47
https://github.com/coreos/fleet/pull/1509/commits/cb8fd41c5f3ef0f223d81bb8828c155cdac6d1b2

So that's why I didn't get those comments, and yes please if you think that there are real issues, then sure we are doing something wrong here ;-)